src/compiler/transformers/module/module.ts

```diff
@@ -1019,7 +1019,7 @@ export function transformModule(context: TransformationContext): (x: SourceFile
         return factory.updateIfStatement(
             node,
             visitNode(node.expression, visitor, isExpression),
-            Debug.checkDefined(visitNode(node.thenStatement, topLevelNestedVisitor, isStatement, factory.liftToBlock)),
+            visitNode(node.thenStatement, topLevelNestedVisitor, isStatement, factory.liftToBlock) ?? factory.createBlock([]),

```

Comment:
alternatively, this could use `factory.createIdentifier("")` - like https://github.com/microsoft/TypeScript/pull/59374 . Producing an empty block seemed slightly better to me here but maybe the goal should be not to alter the input code too much